demisto_sdk/commands/download/downloader.py

```diff
@@ -0,0 +1,510 @@
+        :return: None
+        """
+        file_entity: str = custom_content_object['entity']
+        file_path: str = custom_content_object['path']
+        file_type: str = custom_content_object['type']
+        file_name: str = custom_content_object['name']
+
+        output_path: str = os.path.join(self.output_pack_path, file_entity)
+        output_name: str = os.path.basename(file_path)
+        output_path: str = os.path.join(output_path, output_name)
+        shutil.move(src=file_path, dst=output_path)
+
+        print_color(f'- Added {file_type} "{file_name}"', LOG_COLORS.NATIVE)
+
+    def get_corresponding_pack_content_object(self, custom_content_object: dict) -> dict:
+        """TODO: add unit test
+        Returns the corresponding pack content object to the given custom content object
+        :param custom_content_object: The custom content object to merge into the pack
+        :return: The corresponding pack content object
+        """
+        file_entity: str = custom_content_object['entity']
+        file_name: str = custom_content_object['name']
+        pack_entity_instances: list = self.pack_content[file_entity]
+
+        for pack_entity_instance in pack_entity_instances:
+            if file_name in pack_entity_instance:
+                return pack_entity_instance
+        return {}
+
+    @staticmethod
+    def get_corresponding_pack_file_object(searched_basename: str, pack_content_object: dict) -> dict:
+        """TODO: add unit test
+        Searches for the file named searched_basename under the pack content object and returns it
+        :param searched_basename: The basename to look for
+        :param pack_content_object: The pack content object
+        :return: The pack file object
+        """
+        for _, file_objects in pack_content_object.items():
+            for file_object in file_objects:
+                if file_object['path'].endswith(searched_basename):
+                    return file_object
+        return {}
+
+    def update_data(self, file_path: str, corresponding_pack_file_path: str, file_ending: str) -> None:
+        """TODO: add unit test
+        "Smart" merges the yml/json file with the fields deleted by Demisto when custom content was fetched
+        :param file_path: The file path
+        :param corresponding_pack_file_path: The corresponding pack file path
+        :param file_ending: The file ending
+        :return: None
+        """
+        pack_obj_data, _ = get_dict_from_file(corresponding_pack_file_path)
+        fields: list = DELETED_YML_FIELDS_BY_DEMISTO if file_ending == 'yml' else DELETED_JSON_FIELDS_BY_DEMISTO
+        preserved_data: dict = unflatten({field: dictor(pack_obj_data, field) for field in fields if
+                                          dictor(pack_obj_data, field)}, splitter='dot')
+
+        if file_ending == 'yml':
+            with open(file_path, 'r') as yf:
+                file_yaml_object = self.ryaml.load(yf)
```

Comment:
That might be a good idea, let's leave it for now since the get_yaml function used broadly across the sdk and I don't it to postpone the release. I'll take a look at it the next week. What are you thinking?

Reply:
OK make sure to check that